ostelco-ios-client/Network/APIManager.swift

```diff
@@ -71,3 +71,34 @@ class APIManager: Service {
         }
     }
 }
+
+extension APIManager {
+    
+    #warning("NSError should contain useful information inside onSuccess")
+    func getRegionFromRegions(completion: @escaping (RegionResponse?, Error?) -> Void) {
```

Comment:
Thought we could refactor how we use the APIManager, because everytime we use any resource we need to duplicate the below code.

Reply:
Will you be able to use PromiseKit here?.  The default is always the main queue

Reply:
Mind if I add a warning to add PromiseKit for now? I know it's bad to leave unfinished business, but I want a new test flight version out with these fixes. Right now the one guy in our team who's testing almost everyday is completely stuck 😓 
I'll make sure I fix it afterwards

Reply:
Doesn't need to be a warning, can just be a `// todo`. And this is fine by me

Reply:
I'll do that, and do the refactoring when your branch is merged with the APIHelper

Reply:
I've added todos to refactor into PromiseKit